### ISSUE TITLE
No Configuration files in 1.4.x

### ISSUE THREAD

**Issue by Chara-Reborn** [AlexIIL/BetterLoadingScreen#52]
Reading up on past issues, 1.3.X had at the very least _some_ sort of configuration. How would we go about this configuring the mod for 1.4.X? (Or at least to my knowledge, the latest release?)

**Comment by AlexIIL** [comment]
Currently it's configured with resourcepacks (I should probably add a way to make the config.json file in a main config folder but oh well).

You'll probably want to do all of these steps in a very small modpack for testing purposes, as the only way to view changes is to relaunch the game. So either no other mods or like 5 to make sure that you can open+close the game in 30 seconds or less.

You need to:

- Create a resource pack to store everything in. (And ensure it's loaded - there's a lot more tutorials out there).
- Create a config json file similar to one of them listed [here](https://github.com/AlexIIL/CustomLoadingScreen/tree/1.4.x/src/main/resources/assets/customloadingscreen/sample/config) in `assets/customloadingscreen/config/<name>.json`
- Reference that in the main config file (either add "\<name\>" to `random_configs` or change `screen_config` to \"<name\>" in `config/customloadingscreen.cfg` 

Then try to work out what you can do and change - if you can do that then I'll post more info. (And I'll post this info to the wiki if it works).

**Comment by cosmicdan** [comment]
> (I should probably add a way to make the config.json file in a main config folder but oh well).

Not a big deal. Just recommend on your Curse page that people use the "Resource Loader" mod. If you're not familiar with it, it provides a new "resources" folder that is essentially a forced-loaded resource pack. It's been a stable of mod pack developers for ages.

**Comment by AlexIIL** [comment]
In the 1.7.10 versions I had to add special support for resource loader, because custom loading screen loads *too early* to use resource loader. I haven't ported that support yet or tested resource loader with it so I can't really recommend it properly.

And there's a bigger problem - creating a custom resource file or folder pack is probably a lot easier that trying to follow the non-existent documentation for CLS anyway :P

**Comment by cosmicdan** [comment]
> because custom loading screen loads _too early_ to use resource loader

Good point. I haven't tested it yet (I'm in the middle of writing another issue first) but Resource Loader is 100% a coremod so it SHOULD work. I'll get back to you shorty.

**Comment by cosmicdan** [comment]
Resource Loader seems to work, however the path is not what I expected.

Your mod jar has the config at...
`assets/customloadingscreen/sample/config/default.json`
...and the config  to use this was...
`sample/default`

But I had to make a json at this location to get it to work...
`[instance-folder]/resources/customloadingscreen/config/mymodpack/blank.json`
...with the screen_config setting set to...
`mymodpack/blank`

It kind of makes sense to me (sample is a special prefix that tells your mod to use internal resources or something like that) but yeah. 

So, just an FYI. Ofc I expect you'll want to test with Resource Loader yourself, or you may end up doing your own resource loading anyway.

Cheers!

**Comment by AlexIIL** [comment]
Ok, thanks for testing.

It seems a bit odd that you have to use `config/mymodpack/blank.json`, does `config/blank.json` by itself not work? (Also I should probably look into allowing you to use `mymodpack/config/blank.json` as well somehow).

**Comment by cosmicdan** [comment]
Oh sure, putting the file at `resources/customloadingscreen/config/blank.json` works fine - as long as I set the mod config `S:screen_config=blank`. That makes more sense, it's not necessary to put the jsons in another subfolder.

I love how data-driven the mod is, awesome work! Almost endless possibilities for loading screen designs.
